packages/analytics-browser/src/browser-client.ts

```diff
@@ -26,18 +27,32 @@ export class AmplitudeBrowser extends AmplitudeCore<BrowserConfig> {
       deviceId: oldCookies.deviceId ?? options?.deviceId,
       sessionId: oldCookies.sessionId ?? options?.sessionId,
       optOut: options?.optOut ?? oldCookies.optOut,
+      lastEventTime: oldCookies.lastEventTime,
     });
     await super.init(undefined, undefined, browserOptions);
 
-    // Step 3: Store user session in cookie storage
-    updateCookies(this.config, oldCookies.lastEventTime);
-
-    // Step 4: Install plugins
+    // Step 3: Install plugins
     await this.add(new Context());
     await this.add(new Destination());
 
     // Step 4: Track attributions
-    void this.trackAttributions();
+    if (!options?.attribution?.disabled) {
+      await this.trackCampaign(options?.attribution?.excludeReferrer, options?.attribution?.initialEmptyValue);
+    }
+  }
+
+  async trackCampaign(excludeReferrers?: string[], initialEmptyValue?: string) {
+    const tracker = this.track.bind(this);
+    const onNewCampaign = this.setSessionId.bind(this, Date.now());
```

Comment:
What is this line doing? It looks like it's setting the onNewCampaign which I thought was a boolean to the result of binding the session id to right now...

Reply:
I thought this behavior was going to be controlled by session definition, which is why I ask - is that somehow achieved by this line?

Reply:
this creates a callback function that the "attribution module" will call if a new campaign is found. intention is to keep "attribution module" separate from this amplitude client; allowing to be reused with other amplitude client versions like amplitude-js, if we go that route.